test/infra/proxy.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import asyncio
+from typing import Tuple
+
+import aiotools
+from loguru import logger as LOG
+
+from .async_utils import EventLoopThread, race_tasks
+
+
+class Proxy(EventLoopThread):
+    """
+    A TCP proxy, running as a context manager.
```

Comment:
What are we doing when HTTP/3 is a thing? That would be UDP based. Would we need to maintain two proxy implementations?

Reply:
Does CCF actually have any short/medium term plans on adding HTTP/3? But generally yes, we would probably need a second proxy.

Reply:
This also assumes we even have an HTTP/3 client, which we currently don't.

Reply:
This is not a concern for now but I'm just trying see how much complexity having a proxy in the tests adds.